File: 프로그래머스/1/155652. 둘만의 암호/둘만의 암호.cpp
#include <string>
#include <vector>
#include <algorithm>
using namespace std;

string solution(string s, string skip, int index) {
    string answer = "";
    
    for(int i = 0; i< s.size();i++)
    {
        
        for(int j = 0; j<index;j++)
        {
            s[i]+=1;
            if(s[i] >'z') s[i] = 'a';
            while(find(skip.begin(),skip.end(),s[i])!= skip.end())
            {
            s[i]+=1;
                if(s[i] >'z') s[i] = 'a';
            }
        }
        
        
        answer+= s[i];
    }
    return answer;
}

File: 프로그래머스/1/155652. 둘만의 암호/둘만의 암호.cs
using System;
using System.Linq;
using System.Collections.Generic;
public class Solution {
    public string solution(string s, string skip, int index) {
        string answer = "";
        HashSet<char> skipSet = new HashSet<char>(skip);

string alpha = new string(
    Enumerable.Range('a', 26) 
              .Select(n => (char)n)
              .Where(c => !skipSet.Contains(c))
              .ToArray()
);

foreach(var item in s)
{
    int prev = alpha.IndexOf(item);
    answer += alpha[(prev + index)%(alpha.Length)];
}
        return answer;
    }
}